CMakeLists.txt

```
@@ -99,6 +99,9 @@ if(PNG_FOUND)
   set(ADDITIONAL_LIBRARIES ${ADDITIONAL_LIBRARIES} ${PNG_LIBRARIES})
 endif(PNG_FOUND)
 
+if(UNIX)
```

- [`CMAKE_SYSTEM_NAME`](https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_NAME.html)
- [`UNIX`](https://cmake.org/cmake/help/latest/variable/UNIX.html) is also true on Macos
```suggestion
if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
```

---

Don't you want/need this on macOS as well? Certainly things like FreeBSD and friends should have it too. Maybe `UNIX AND NOT APPLE` perhaps?

---

I don't own any Apple things myself, but I remember somebody saying that linking to the `m` library is not needed there.
See this stackoverflow: https://stackoverflow.com/questions/33675638/gcc-link-the-math-library-by-default-in-c-on-mac-os-x
Your suggestion looks good.

---

So, would you still prefer I modified `if(UNIX)` to `if(UNIX AND NOT APPLE)` or is the PR ok as it is?

---

I don't know about the BSD's, I know the `m` library is ok on Macos (but not needed).
So I think it's ok.
There is no ci to confirm it though.